src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1160,6 +1162,53 @@ static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
 
 } // namespace BinaryIntrinsicFunction
 
+
+namespace ObjectType {
+
+     static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args != 1,
+            "ASR Verify: type() takes only 1 argument `object`",
+            x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_ObjectType(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*>& args) {
+        std::string object_type = "<type '" + ASRUtils::type_to_str_python(t1) + "'>";
+        return StringConstant(object_type, character(object_type.length()));
+    }
+
+    static inline ASR::asr_t* create_ObjectType(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() > 1) {
+            err("type() takes only 1 argument `object`", loc);
+        }
+
+        Vec<ASR::expr_t *> arg_values;
+        arg_values.reserve(al, 1);
+        arg_values.push_back(al, expr_value(args[0]));
+        ASR::expr_t *m_value = eval_ObjectType(al, loc, expr_type(args[0]), arg_values);
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::ObjectType),
+            args.p, args.n, 0, ASRUtils::expr_type(args[0]), m_value);
+    }
+
+    static inline ASR::expr_t* instantiate_ObjectType(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_objectType_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("a", arg_types[0]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+
+        ASR::symbol_t *f_sym = make_ASR_Function_t(fn_name, fn_symtab, dep, args,
+            body, result, ASR::abiType::Source, ASR::deftypeType::Implementation, nullptr);
+        scope->add_symbol(fn_name, f_sym);
+        return b.Call(f_sym, new_args, return_type, nullptr);
+    }
```

Comment:
I think the compile time value will always be used, so there is no need for instantiate function.
```suggestion
```